sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -918,14 +1082,28 @@ public void validate(PCollection<T> input) {
 
       // Require a schema if creating one or more tables.
       checkArgument(
-          getCreateDisposition() != CreateDisposition.CREATE_IF_NEEDED || getJsonSchema() != null,
+          getCreateDisposition() != CreateDisposition.CREATE_IF_NEEDED
+              || getJsonSchema() != null || getDynamicDestinations() != null
+          || getSchemaFromView() != null,
           "CreateDisposition is CREATE_IF_NEEDED, however no schema was provided.");
 
+      checkArgument(getJsonTableRef() == null || getTableFunction() == null,
```

Comment:
Icky. Probably better do something like: `checkArgument(1 == Iterables.size(Iterables.filter(new Object[] {getJsonTableRef(), getTableFunction(), getDynamicDestinations()}, Predicates.nonNull())), "Exactly one of ..., ..., ... must be set")`